dipy/direction/peaks.py

```diff
@@ -157,8 +157,63 @@ def peak_directions(odf, sphere, relative_peak_threshold=.5,
     return directions, values, indices
 
 
+def _rebuild_pam(sphere, peak_indices, peak_values, peak_dirs,
```

Comment:
It would be a bit weird as a class method, considering that it returns another instance of that class. Or would it? 

There does seem to be an opportunity to reduce boiler-plate here though, so I am going to try out a couple more things. Stand by.